advanced_alchemy/types/file_object/backends/fsspec.py

```diff
@@ -0,0 +1,404 @@
+                all_data = b"".join([chunk async for chunk in data])
+                return await async_(self.save_object)(file_object=file_object, data=all_data, chunk_size=chunk_size)
+            return await async_(self.save_object)(file_object=file_object, data=data, chunk_size=chunk_size)  # type: ignore
+
+        fs = cast("AsyncFileSystem", self.fs)
+        path_str = self._to_path(file_object.path)
+        size: Optional[int] = None
+
+        # Let type inference work for f, add ignore for context manager protocol errors
+        async with fs.open(path_str, "wb") as f:  # pyright: ignore
+            if isinstance(data, bytes):
+                await f.write(data)  # pyright: ignore
+                size = len(data)
+            elif isinstance(data, AsyncIterator):
+                size = 0
+                async for chunk in data:
+                    await f.write(chunk)  # pyright: ignore
+                    size += len(chunk)
+            elif isinstance(data, (Iterator, Iterable)):  # Sync iterables
+                size = 0
+                for chunk in data:
+                    await f.write(chunk)  # pyright: ignore
+                    size += len(chunk)
+            elif isinstance(data, IO):  # Sync IO
+                size = 0
+                while True:
+                    chunk = await async_(data.read)(chunk_size)  # pyright: ignore
+                    if not chunk:
+                        break
+                    await f.write(chunk)  # pyright: ignore
+                    size += len(chunk)  # pyright: ignore
+            elif isinstance(data, Path):
+                if not data.is_file():
+                    raise FileNotFoundError(data)
+                size = data.stat().st_size
+                try:
+                    import aiofiles
+
+                    async with aiofiles.open(data, mode="rb") as source_f:
+                        while True:
+                            chunk = await source_f.read(chunk_size)
+                            if not chunk:
+                                break
+                            await f.write(chunk)  # pyright: ignore
+                except ImportError:
+                    with data.open("rb") as source_f:
+                        while True:
+                            chunk = await async_(source_f.read)(chunk_size)
+                            if not chunk:
+                                break
+                            await f.write(chunk)  # pyright: ignore
+            else:
+                msg = f"Unsupported data type: {type(data)}"
+                raise TypeError(msg)
+
+        # Initialize file_object properties with defaults or calculated values
+        file_object.size = size or 0
+        file_object.last_modified = None
+        file_object.etag = None
+        original_metadata = file_object.metadata  # Keep original metadata unless backend provides updates
+
+        info = await fs.info(path_str)  # pyright: ignore
```

Comment:
```suggestion
        info = await fs._info(path_str)  # pyright: ignore
``` use the actual async call